examples/notebooks/03_model_scoring.py

```diff
@@ -0,0 +1,200 @@
+# %% [markdown]
+"""
+# Scoring Models in Earth-2 MIP
+
+The following notebook will demonstrate how to use Earth-2 MIP to perform a scoring
+workflow to assess the accuracy of AI models using ERA5 reanalysis data as the
+ground truth. This can then be extended to score custom models placed into the model
+registry. This tutorial also covers details about using a HDF5 datasource, the expected
+format this data should be formatted and how to use H5 files for evaluating models over
+a year's worth of data.
+
+In summary this notebook will cover the following topics:
+
+- Implementing a basic scoring workflow in Earth-2 MIP
+- HDF5 datasource and the expected data format of the H5 files
+ 
+"""
+# %%
+import os
+import xarray
+import earth2mip
+import datetime
+import numpy as np
+
+# %% [markdown]
+"""
+## Setting up HDF5 data
+
+The first step of scoring is handling the target data. One could simply use the CDSDatasource
+to download target data on the fly, but depending on how comprehensive this can prove quite
+slow. Additionally, many scoring pipelines require on-prem data. So we will demonstrate
+how to use the HDF5 datasource.
+
+The HDF5 data source assumes that the data to be loaded is stored in the general form:
+
+year.h5
+ | - field (time, channels, grid)
+
+So for AFNO which requires 34 channels with a time-step size of 6 hours, an H5 file will
+have the following form of data:
+
+2017.h5
+ | - field (1460, 34, 721, 1440)
+2016.h5
+ | - field (1464, 34, 721, 1440)
+2015.h5
+ | - field (1460, 34, 721, 1440)
+
+One option to build these H5 files from scratch is to use the ERA5 mirror scripts provided
+in [Modulus](https://github.com/NVIDIA/modulus/tree/main/examples/weather/dataset_download).
+(use the input channels defined in each model file to act as a reference about what channels are needed).
+For the rest of this tutorial, it is assumed that 2017.h5 is present for the full year.
+"""
```

Comment:
This made me realize that we do really need to establish clear docs / dictionary of variable names.

Reply:
personally would be nice to define a string encoding for variable names that can be parsed into the needed information (level type, level, ECWMF param ID)...sort of like a url for variable names. then we can add a python tool for parsing these strings (as cds.parse_channel) does now.